### DISCUSSION TITLE
error in cache directory/file

### DISCUSSION THREAD

**Discussion by sgsalmo** [h-a-graham/rayvista#6] [Q&A] [answered]
Hi Hugh,

I tried to follow your code posted in twitter. But then, I got an error message, as:

<img width="570" alt="3d error message_rayvista" src="https://user-images.githubusercontent.com/80515793/118618998-c59a6580-b7f6-11eb-90fc-3a41517dea61.png">

I am not sure what gives the error. Hope you can help me again. I am using Macbook Pro (on Mac OS Big Sur), on RStudio 1.4.1103. 

One of your suggestions is to make a "file path". Sorry, Im not that familiar with this. Hope you can help clarify.

Thanks,

SS

**Comment by h-a-graham** [comment, marked as answer]
Here is the function with an example of how to specify and create the full file path: 

```
#define full directory path
my_cache <-  '/home/sgsalmo/rayvistaproject/cache'

#check if directory exists - if not then this creates a new directory
if (!file.exists(my_cache)){
    dir.create(my_cache)
}

plot_mountain <- function(.coord, .name, .filt){
  next3d()
  mountain <- plot_3d_vista(lat=.coord[1], long=.coord[2], radius = 7500,
                            zscale=3, elevation_detail = 14, overlay_detail= 14,
                            outlier_filter = .filt ,zoom=0.8,
                            theta=sample(-175:-1, 1), windowsize =c(800,1200),
                            phi=15, solid=F, cache_dir =my_cache)
  render_label(heightmap= mountain, text=.name, lat = .coord[1],
               long=.coord[2], extent = attr(mountain, 'extent'), relativez=T,
               zscale=3, z=1000, alpha=0, family='mono')
}
```

**Reply by sgsalmo** [reply]
Thanks, Hugh! It is now working.

**Reply by h-a-graham** [reply]
Great, thanks for letting me know!
